Oracle.md

```diff
@@ -113,12 +113,30 @@ If the rate is greater than $999999, the oracle signs the digits `9`, `9`, `9`,
 
 ### Serialization and signing of outcome values
 
-Every outcome value should be encoded as [a UTF-8 string with NFC normalization](./Messaging.md#Fundamental-types), before being passed to the signing algorithm.
+Every outcome value (`outcome_value`) should be encoded as [a UTF-8 string with NFC normalization](./Messaging.md#Fundamental-types), before being passed to the signing algorithm.
 UTF-8 is chosen as being a widely supported and easy to implement encoding format.
 
 For numerical outcomes represented in bases greater than 10, each digit should be converted to base 10 before being encoded (note that base 10 numbers in UTF-8 take values in the 0x0030-0x0039 range).
 This helps preventing any confusion about the capitalization of letters or the introduction of non-standard characters.
 
+#### Signing Algorithm
+
+Signatures should be generated following the algorithm specified in [BIP 340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki).
+Tagged hashes should also be used as defined in the [design section of BIP 340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#Design).
+
+The secret key (`sk`) should be the private key corresponding to the oracle's public key.
+
+The nonce (`R`) should be the nonce the oracle pre-committed to for this event, with `k'` being `R = k'⋅G`.
+
+The message (`m`) should be calculated as follows:
+
+ * Let `tag` = Event descriptor name (eg a Version 0 enum_event_descriptor is `enum_event_descriptor_v0`)
+ * Let `outcome_value` = the serialized event outcome value being attested to by the oracle
+
+Message (`m`) will be: hash <sub>tag</sub> (R || outcome_value)
+
+Finally, the signature should satisfy the [BIP 340 Verification algo](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#verification).
+
```

Comment:
Some questions:

- Is this only for signing the attestation? What about the announcement?
- Why is there R in the attestation message?

What do you thing about the following idea:

tags should separate your signatures from other *applications* not from different uses of signatures in the same application. So we could just use `oracle` (it would be nice to have spec numbers so we could be self-referential like BIP340) and then differentiate between announcement and attestation within the message itself.

I would also prefer to be explicit about the signing and verification algorithms. e.g

The signing algorithm `sign(sk, message)` takes as input:

- *sk*: a secp256k1 secret key
- *message*: the message to be signed as an array of bytes

and is defined as:

- Set *m* to hash<sub>oracle</sub>(*message*)
- Return BIP340_sign(*sk*, *m*)

Reply:
in case anyone is confused, I think #111 is the issue